resolver/src/p2p/ingress.rs

```diff
@@ -88,3 +101,70 @@ impl<K: Span> Resolver for Mailbox<K> {
             .expect("Failed to send cancel_all");
     }
 }
+
+impl<K: Span, P: PublicKey> Mailbox<K, P> {
+    /// Send a fetch request with initial hints for which peers likely have the data.
+    ///
+    /// Hinted peers are tried first. If a hinted peer fails (timeout, error response,
+    /// or send failure), only that peer is removed from hints. As hints deplete through
+    /// failures, the resolver naturally falls back to trying all peers.
+    ///
+    /// Hints are automatically cleared when:
+    /// - The fetch succeeds (data received)
+    /// - The fetch is canceled
+    /// - All hinted peers fail (empty hints trigger fallback to all peers)
+    ///
+    /// Panics if the send fails.
+    pub async fn fetch_hinted(&mut self, key: K, hints: Vec<P>) {
```

Comment:
Wondering whether these methods should be added to the `Resolver` trait or not.

Reply:
IMO, no; This seems fairly specific to the p2p backend. The exoware backend won't need this concept.